vortex-array/src/compute/search_sorted.rs

```diff
@@ -19,7 +22,14 @@ pub fn search_sorted<T: Into<Scalar>>(
     side: SearchSortedSide,
 ) -> VortexResult<usize> {
     let scalar = target.into().cast(array.dtype())?;
-    array
+    if let Some(search_sorted) = array.search_sorted() {
+        return search_sorted.search_sorted(&scalar, side);
+    }
+
+    // Otherwise, flatten and try again.
```

Comment:
I'm less sure about this one. What if the array is e.g. a struct array? Do we really want to flatten and then fail on search_sorted? Maybe we should gate this DType so that we can guarantee the flattening isn't for nothing